### PR TITLE
Added compounds with matching formulas to the Compounds autofill

### LINKED ISSUE #1153: Cannot add compound synonym if the compound isn't present
@mneinast - So I think I can accurately guess what's going on here.  Here's what I expect happened.

- When you ran the start page to autofill, the compound synonym didn't exist.
- The compound existed, but it wasn't autofilled in the compounds sheet because the synonym search failed to find it
- So the autofill added a partial record based on the information it got from the peak annotation file (the synonym and the formula).  It created new rows for that data and could not fill in the HMDB ID nor the synonyms
- Then, when you went to validate, having added the HMDB ID, (which is required to be unique), you ended up with a conflicting value error.
- the solution is to add the synonyms from the Peak Annotations file to the existing compound records as synonyms instead of as separate compound records.
- The problem is that the autofill doesn't provide that compound record for you to edit.

I think that the solution here is to auto-populate all existing compounds and hope that the user will see that the autofilled row for what is essentially a compound synonym of another row and delete that row and add the synonym to the other record.

I'll create an issue for this.

_Originally posted by @hepcat72 in https://github.com/Princeton-LSI-ResearchComputing/tracebase/issues/1049#issuecomment-2289194431_

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

The issue mentions adding "all compounds", but given Michael's concern about identifying the compounds from the peak annotation files and the large number of compounds in tracebase, I decided to only add those with matching formulas, which should both be easier for the researcher to pour through and remain scalable as tracebase grows.

That also solves the original problem, which was there was no way to add a missing synonym if the compound to add it to wasn't present in the sheet.

## Affected Issues/Pull Requests

- Resolves #1153
- Merges into main

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/views/upload/submission.py [hunk 1]
```diff
@@ -1567,6 +1568,25 @@ def extract_autofill_from_peak_annotation_files(self):
                         CompoundsLoader.DataHeaders.FORMULA: formula,
                     }
 
+            print("EXTRACTING AND INCLUDING ALL COMPOUNDS WITH MATCHING FORMULAS")
```
Comment (hepcat72) [inline DataRepo/views/upload/submission.py:1571]: C. Still using the debugs for manual testing.

File: DataRepo/views/upload/submission.py [hunk 2]
```diff
@@ -1567,6 +1568,25 @@ def extract_autofill_from_peak_annotation_files(self):
...
+            for formula in np.unique(formulas):
+                if formula in self.none_vals:
+                    continue
+                for compound_rec in Compound.objects.filter(formula__iexact=formula):
+                    if (
+                        compound_rec.name
+                        not in self.autofill_dict[CompoundsLoader.DataSheetName].keys()
+                    ):
```
Comment (hepcat72) [inline DataRepo/views/upload/submission.py:1579]: C. Don't add the compound if it was already added.

**Comment by hepcat72** [comment]
b. Working with Michael's "water" submission data, it appears that grabbing any compound by formula is not enough.  The formulas from the peak annotation file can differ from the documentaed compound due to ionization.  E.g. This error:

```
ConflictingValueErrors
Conflicting values encountered during loading:
	During the processing of sheet [Compounds] in MASTER STUDY DOC FOR TRACEBASE_exp011b3_240821_add HMDB IDs-edit.xlsx...
		File record:     {'name': 'Indole-3-acetamide', 'formula': 'C10H10N2O', 'hmdb_id': 'HMDB0029739'} (on row(s): 228)
		Database record: {'name': '1H-indole-3-acetamide', 'formula': 'C10H9NO2', 'hmdb_id': 'HMDB0029739'}
			[name] values differ:
			- database: [1H-indole-3-acetamide]
			- file:     [Indole-3-acetamide]
			[formula] values differ:
			- database: [C10H9NO2]
			- file:     [C10H10N2O]
```
reveals that existing compound `1H-indole-3-acetamide` was not included in the downloaded template because it differs by a proton and the synonym doesn't exist.

**Review by lparsons — APPROVED** [review]
Seems like a reasonable solution, at least for now. Might want to remove the print statements, or put them behind some debug conditional to keep the web logs from getting cluttered.

**Pull request merged**